### ISSUE TITLE
feat[dev]: ✨ implement A11y improvements to sd-badge

### ISSUE THREAD

**Issue by yoezlem** [solid-design-system/solid#1466]
## Description
This task describes the Storybook implementation for the sd-badge component as part of the A11y improvements outlined in Epic #1465.

A11y improvements for sd-badge can be found here: [sd-badge Notes](https://www.notion.so/matuzo/sd-badge-103f3f6e1bc080619981e3e8d9546a9f).

------------------------------------------

## Comment
**Problem:** In the docs, it says that the success and error variants don't imply any semantic meaning. They kinda do because they tell me that something good, bad, or neutral is going on. You can argue that the number itself is enough to communicate that information, at least as a fallback, but that's not the case in the sm variation. It only communicates the state visually.
**Solution:** 
- [x] Use the `.sr-only` class instead of `.hidden` for the content part in sm variations.
- [ ] Extend the hint to make clear why we have the variant names and referr to the ticket (TBA) -> to be addressed in [this ticket](https://github.com/solid-design-system/solid/issues/1593)

**Problem:** Badges are focusable. That can become a problem when they are nested inside other interactive elements. When you use the Tab key to navigate, the JAWS screen reader, for example, only announces "Label, button". When you Tab again, the focus jumps to the next element, the badge, and JAWS announces "8". That's not ideal.

```
<sd-badge size="lg" variant="default">
	#shadow-root
	<span tabindex="0" role="status" part="base" class="inline-flex items-center justify-center gap-x-[1px] text-center leading-none whitespace-nowrap border rounded-full select-none cursor-[inherit] text-white bg-primary-500 border-white h-5 px-[5px] min-w-[20px] text-[12px]">
    <span part="content" class="">
      <slot></slot>
    </span>
  </span>

8
</sd-badge>
```
**Solution:** Remove the `tabindex`, at least when badges are nested inside links, buttons, or other interactive elements.

**Problem:** 
The badge is a live region. I assume it's to announce live updates. Please be aware that updating just the number without any text can be confusing because screen reader users will just hear a seemingly random number. 
**Solution:** Add a new property `context` to the badge (name can be discussed).
```html
<sd-badge size="lg" variant="default" context="new messages">8</sd-badge>
```

- [x] fix #1537 

## Notes (aaded to the DaDs)
- [x] The thinness of the font makes content hard to read, especially in md variations => prevent using the md-variants
- [x] The font size is defined in px, an absolute unit. Consider using rem for font sizes. => tackled in / blocked by #1546 

## Best Practices
If the badge communicates critical information, e.g., the number of items in a cart, ensure it's not the only place users can obtain it. For example, in the cart itself, the number should be repeated in text.

## Related WCAG criteria
https://www.w3.org/WAI/WCAG22/quickref/#name-role-value

### Open Questions
- [ ] Question1
- [ ] Question2

## DoR
- [x] Item has been estimated by the team
- [x] Item dependencies have been identified and documented

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [x] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [x] Relevant stories (Features, A11y) are created/updated
- [x] Implementation works successfully on `feature` branch

**Comment by yoezlem** [comment]
For Comments inside Notion please check the Notion page!

**Comment by smfonseca** [comment]
> Problem:
The badge is a live region. I assume it's to announce live updates. Please be aware that updating just the number without any text can be confusing because screen reader users will just hear a seemingly random number.
Solution: Add a new property context to the badge (name can be discussed).

After discussing this with @mariohamann we agreed not to implement the proposed solution. Adding this would mean having to correctly handle the order in which screen readers announce the information (language dependent) and this would increase the component complexity. 

A more flexible approach is for users to include a hidden span with `.sr-only` which includes the content that needs to be announced. We will include this as a hint in the documentation.

**Comment by smfonseca** [comment]
@MartaPintoTeixeira @coraliefeil the task _The thinness of the font makes content hard to read, especially in md variations => prevent using the md-variants_ requires input for design. Please let me know once you have a decision on how to proceed. Thank you

**Comment by MartaPintoTeixeira** [comment]
> @MartaPintoTeixeira @coraliefeil the task _The thinness of the font makes content hard to read, especially in md variations => prevent using the md-variants_ requires input for design. Please let me know once you have a decision on how to proceed. Thank you

<img width="200" alt="Screenshot 2024-10-17 at 15 16 36" src="https://github.com/user-attachments/assets/a134670f-5ae5-4c4f-9a99-3daa201520b1">

My thoughts on this:
- if we change the font weight from in "md" we should also change it in "lg"
- regarding readability I made a quick research and apple iphones for ex use regular, nevertheless is true that bold can help readability on smaller sizes and for this reason I would consider changing copy to bold.

I will prepare a branch with bold fonts for "lg" and "md" and in the meantime we wait for Coralies feedback before changing storybook.

**Comment by MartaPintoTeixeira** [comment]
@smfonseca I have aligned with Coralie. Lets change "lg" and "md" fonts to bold.
I have created a ticket to change map-maker font to bold as the font size is the same size.

**Comment by smfonseca** [comment]
Font weight for ´lg´ and ´md´ variants updated in the [current PR](https://github.com/solid-design-system/solid/pull/1563)

**Comment by MartaPintoTeixeira** [comment]
Font weight in badge approved
